cmd/odf/restore/crds.go

```diff
@@ -1,12 +1,55 @@
 package restore
 
 import (
+	"fmt"
+	"slices"
+	"strings"
+
 	"github.com/red-hat-storage/odf-cli/cmd/odf/root"
 	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
 	pkgrestore "github.com/rook/kubectl-rook-ceph/pkg/restore"
 	"github.com/spf13/cobra"
 )
 
+// groupVersions defines the supported CRD groups and their corresponding API versions.
+var groupVersions = map[string]string{
+	"ocs.openshift.io":       "v1",
+	"ceph.rook.io":           "v1",
+	"csi.ceph.io":            "v1alpha1",
+	"odf.openshift.io":       "v1alpha1",
+	"noobaa.io":              "v1alpha1",
+	"csiaddons.openshift.io": "v1alpha1",
+}
+
+// groupNameKeys returns the keys of a string map. It is used to print out supported group names.
+func groupNameKeys(m map[string]string) []string {
+	out := make([]string, 0, len(m))
+	for k := range m {
+		out = append(out, k)
+	}
+	return out
+}
+
+// parseFullyQualifiedCRD takes a fully qualified CRD type of the form "resource.group"
+// (for example, "cephclusters.ceph.rook.io") and returns the resource name, group name, and
+// the API version associated with that group. It returns an error if the format is invalid
+// or the group is not recognized.
+func parseFullyQualifiedCRD(fqcrd string) (resourceName, groupName, version string, err error) {
+	parts := strings.SplitN(fqcrd, ".", 2)
+	if len(parts) != 2 {
+		return "", "", "", fmt.Errorf("invalid CRD format %q; expected format <resource>.<group>", fqcrd)
+	}
+	resourceName = parts[0]
+	groupName = parts[1]
+
+	version, ok := groupVersions[groupName]
+	if !ok {
+		return "", "", "", fmt.Errorf("unsupported group %q; supported groups are: %v", groupName, groupNameKeys(groupVersions))
+	}
+	return resourceName, groupName, version, nil
+}
+
```

Comment:
any reason in not using the established GVK/GVR schema from https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/schema/group_version.go or https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/group_version.go if we want a higher level of abstraction.

is it like you don't want to import Go Structs from other repos? as of now iiuc, we already made an effort moving APIs into a submodule.

Reply:
@leelavg The input format is `./bin/odf restore deleted storageclusters.ocs.openshift.io`, where only the CR name and group are provided (storageclusters.ocs.openshift.io). We do not specify the version. Please add the relevant code to improve the current implementation.